commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -418,102 +318,182 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
         return populateTax(resultsFromStaticResponse);
     }
 
-  // Structure to store the tax data retrieved from external service. This class simplifies our
-  // ability to access the data when storing it in Salesforce's CartTaxDto.
-  class TaxDataFromExternalService {
-    private Decimal rate;
-    private Decimal amount;
-    private String taxName;
-    private Decimal adjustmentTaxAmount;
-    private Decimal totalItemizedPromotionTaxAmount;
-    private List<CartAdjustment> itemizedPromotionTaxAmounts;
-    private Decimal grossUnitPrice;
-    private Decimal netUnitPrice;
-
-    public TaxDataFromExternalService() {
-      rate = 0.0;
-      amount = 0.0;
-      taxName = '';
-      adjustmentTaxAmount = 0.0;
-      totalItemizedPromotionTaxAmount = 0.0;
-      itemizedPromotionTaxAmounts = null;
-      grossUnitPrice = 0.0;
-      netUnitPrice = 0.0;
-    }
 
-    public TaxDataFromExternalService(
-      Decimal rateObj,
-      Decimal amountObj,
-      String taxNameObj,
-      Decimal adjustmentTaxAmountObj,
-      Decimal totalItemizedPromotionTaxAmountObj,
-      List<CartAdjustment> itemizedPromotionTaxAmountsObj,
-      Decimal grossUnitPriceObj,
-      Decimal netUnitPriceObj
-    ) {
-      rate = rateObj;
-      amount = amountObj;
-      taxName = taxNameObj;
-      adjustmentTaxAmount = adjustmentTaxAmountObj;
-      totalItemizedPromotionTaxAmount = totalItemizedPromotionTaxAmountObj;
-      itemizedPromotionTaxAmounts = itemizedPromotionTaxAmountsObj;
-      grossUnitPrice = grossUnitPriceObj;
-      netUnitPrice = netUnitPriceObj;
+	private boolean isValidCart(CartExtension.Cart cart){
+		CartExtension.CartDeliveryGroupList cartDeliveryGroups = cart.getCartDeliveryGroups();
+		if(cartDeliveryGroups == null || cartDeliveryGroups.size() == 0){
+			return false;
+		}
+
+		CartExtension.CartDeliveryGroup cartDeliveryGroup = cartDeliveryGroups.get(0);
```

Comment:
We should process cases then there is more than one delivery group.

Reply:
As discussed, this class only considers first delivery group